src/java/org/apache/cassandra/service/accord/txn/TxnCondition.java

```diff
@@ -289,54 +291,99 @@ public boolean applies(TxnData data)
+                            // This is NULL when touched, so need to still check each value
+                            if (exists)
+                            {
+                                CollectionType<?> type = (CollectionType<?>) reference.column().type.unwrap();
+                                switch (type.kind)
+                                {
+                                    case MAP:
+                                    {
+                                        exists = !type.nameComparator().isNull(cell.path().get(0));
+                                        if (exists)
+                                            exists = !type.valueComparator().isNull(cell.buffer());
+                                    }
+                                    break;
+                                    case SET:
+                                    {
+                                        exists = !type.nameComparator().isNull(cell.path().get(0));
+                                    }
+                                    break;
+                                    case LIST:
+                                    {
+                                        exists = !type.valueComparator().isNull(cell.buffer());
+                                    }
+                                    break;
+                                    default:
+                                        throw new UnsupportedOperationException(type.kind.name());
+                                }
+                            }
+                        }
+                        else if (reference.isFieldSelection())
+                        {
+                            Cell<?> cell = (Cell<?>) columnData;
+                            exists = exists(cell, reference.getFieldSelectionType());
+                        }
+                        else
+                        {
+                            // TODO: Is this even necessary, given the partition is already filtered?
+                            if (!((ComplexColumnData) columnData).complexDeletion().isLive())
+                                exists = false;
+                        }
+                    }
+                    else if (reference.isElementSelection())
+                    {
+                        // This is frozen, so check if the Cell is a tombstone and that the element is present.
+                        Cell<?> cell = (Cell<?>) columnData;
+                        exists = exists(cell, reference.column().type);
+                        if (exists)
+                        {
+                            ByteBuffer element = reference.getFrozenCollectionElement(cell);
+                            exists = !reference.getFrozenCollectionElementType().isNull(element);
+                        }
+                    }
+                    else if (reference.isFieldSelection())
```

Comment:
im not sure if this is allowed in CQL, but it was already here in the code so i made sure it worked